File: src/main/java/info/szadkowski/contact/properties/ThrottlingProperties.java
package info.szadkowski.contact.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.time.Duration;

@Configuration
@ConfigurationProperties(prefix = "throttling")
@Data
public class ThrottlingProperties {
  private Duration clearExpiredRate;
  private ThrottlingScope ip;
  private ThrottlingScope all;

  @Data
  public static class ThrottlingScope {
    private long limit;
    private Duration window;
  }
}

File: src/main/kotlin/info/szadkowski/contact/properties/ThrottlingProperties.kt
package info.szadkowski.contact.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import java.time.Duration

@ConstructorBinding
@ConfigurationProperties(prefix = "throttling")
data class ThrottlingProperties(
    val clearExpiredRate: Duration,
    var ip: ThrottlingScope,
    var all: ThrottlingScope
) {
    data class ThrottlingScope(
        val limit: Long,
        val window: Duration
    )
}
